### PR TITLE
Add matching return type for search method

### PR DESCRIPTION
**Pull request by Joelmmm**

### CODE REVIEW

File: src/Endpoint.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export class Endpoint implements IEndpoint {
   base: string;
   availableFunctions: Array<string>;
   request: (params: Object) => Promise<any>;
-  search: (query: string, filters: object) => Function;
+  search: (query: string, filters: object) => Promise<Object>;
```
Comment (PorterK) [inline src/Endpoint.ts]: Thanks for the PR - is `Promise<unknown>` more appropriate here?

**Review by PorterK — COMMENTED** [review]
Sorry this took so long to get to me! Github inbox is forever long

File: src/Endpoint.ts (hunk 1 above)
Comment (Joelmmm) [reply src/Endpoint.ts]: Yes it's a better solution

**Comment by PorterK** [comment]
@Joelmmm Thanks for the contribution! This has been released under v4.0.1

**Pull request merged**